web/src/test/java/se/inera/intyg/rehabstod/service/sjukfall/GetRekoStatusServiceTest.java

```diff
@@ -50,56 +51,68 @@ public class GetRekoStatusServiceTest {
-            getRekoStatusService.get(PATIENT_ID, END_DATE, START_DATE);
-            verify(intygstjanstRestIntegrationService).getRekoStatus(captor.capture());
-            assertEquals(END_DATE, captor.getValue().getEndDate());
-        }
-
-        @Test
-        void shouldSetStartDate() {
-            final var captor = ArgumentCaptor.forClass(GetRekoStatusRequestDTO.class);
-            getRekoStatusService.get(PATIENT_ID, END_DATE, START_DATE);
-            verify(intygstjanstRestIntegrationService).getRekoStatus(captor.capture());
-            assertEquals(START_DATE, captor.getValue().getEndDate());
-        }
-    }
+     @Nested
+     class HasRekoStatus {
+
+         @BeforeEach
+         void setup() {
+             when(intygstjanstRestIntegrationService.getRekoStatus(any())).thenReturn(
+                 new RekoStatusDTO(
+                     new RekoStatusTypeDTO(REKO_ID, REKO_NAME)
+                 )
+             );
+         }
+
+         @Nested
+         class TestItResponse {
+
+             @Test
+             void shouldReturnRekoStatusId() {
+                 final var response = getRekoStatusService.get(PATIENT_ID, END_DATE, START_DATE);
+                 assertEquals(REKO_ID, response.getStatus().getId());
+             }
+
+             @Test
+             void shouldReturnRekoStatusName() {
+                 final var response = getRekoStatusService.get(PATIENT_ID, END_DATE, START_DATE);
+                 assertEquals(REKO_NAME, response.getStatus().getName());
+             }
+         }
+
+         @Nested
+         class TestITRequest {
+
+             @Test
+             void shouldSetPatientId() {
+                 final var captor = ArgumentCaptor.forClass(GetRekoStatusRequestDTO.class);
+                 getRekoStatusService.get(PATIENT_ID, END_DATE, START_DATE);
+                 verify(intygstjanstRestIntegrationService).getRekoStatus(captor.capture());
+                 assertEquals(PATIENT_ID, captor.getValue().getPatientId());
+             }
+
+             @Test
+             void shouldSetEndDate() {
+                 final var captor = ArgumentCaptor.forClass(GetRekoStatusRequestDTO.class);
+                 getRekoStatusService.get(PATIENT_ID, END_DATE, START_DATE);
+                 verify(intygstjanstRestIntegrationService).getRekoStatus(captor.capture());
+                 assertEquals(END_DATE, captor.getValue().getEndDate());
+             }
+
+             @Test
+             void shouldSetStartDate() {
+                 final var captor = ArgumentCaptor.forClass(GetRekoStatusRequestDTO.class);
+                 getRekoStatusService.get(PATIENT_ID, END_DATE, START_DATE);
+                 verify(intygstjanstRestIntegrationService).getRekoStatus(captor.capture());
+                 assertEquals(START_DATE, captor.getValue().getEndDate());
+             }
+         }
+     }
+
+     @Nested
+    class HasNoRekoStatus {
```

Comment:
If its expected that we return a ```new RekoStatusDTO()``` if the patient dont have any rekostatus, we could add a test for that aswell so that we know this logic should not be changed in the future!